source/arm9/libc/filesystem.c

```diff
@@ -652,3 +652,38 @@ int symlink(const char *target, const char *path)
     return -1;
 }
 
+int FAT_getAttr(const char *file)
+{
+    if (nitrofs_use_for_path(file))
+        return nitrofs_fat_get_attr(file);
+
+    FILINFO fno = { 0 };
+    FRESULT result = f_stat(file, &fno);
+
+    if (result != FR_OK)
+    {
+        errno = fatfs_error_to_posix(result);
+        return -1;
+    }
+
+    return fno.fattrib;
+}
+
+int FAT_setAttr(const char *file, uint8_t attr)
+{
+    if (nitrofs_use_for_path(file))
+        return -1;
```

Comment:
Does `libfat` actually use `errno` for the error codes?

Either way, if we're using `errno`, it might be a good idea to set it to something here as well.

Reply:
To clarify: I don't think we should aim to be quirk-for-quirk compatible with `libfat`; this seems like a more consistent/harder to get wrong approach. I was just curious.

Reply:
It doesn't, I think? But just getting "-1" is super annoying if you're trying to debug something, and it's essentially free to support errno in our code, so we might as well do it.

Reply:
Yes, in that case it's best to set a sensible `errno` for all error paths.

Reply:
Added.